src/llir/raise/early.rs

```diff
@@ -238,26 +245,49 @@ fn decode_args_with_abi(
                 ScalarValue::Int(extra_arg as _)
             },
 
-            | ArgEncoding::Integer { arg0: false, size: 4, ty_color: _ }
-            | ArgEncoding::Color
             | ArgEncoding::JumpOffset
             | ArgEncoding::JumpTime
-            | ArgEncoding::Padding
+            | ArgEncoding::Integer { arg0: false, size: 4, format: ast::IntFormat { unsigned: false, radix: _ }, .. }
             => {
                 decrease_len(emitter, &mut remaining_len, 4)?;
-                ScalarValue::Int(args_blob.read_u32().expect("already checked len") as i32)
+                ScalarValue::Int(args_blob.read_i32().expect("already checked len") as i32)
             },
 
-            | ArgEncoding::Integer { arg0: false, size: 2, ty_color: _ }
+            | ArgEncoding::Integer { arg0: false, size: 2, format: ast::IntFormat { unsigned: false, radix: _ }, .. }
             => {
                 decrease_len(emitter, &mut remaining_len, 2)?;
                 ScalarValue::Int(args_blob.read_i16().expect("already checked len") as i32)
             },
 
+            | ArgEncoding::Integer { arg0: false, size: 1, format: ast::IntFormat { unsigned: false, radix: _ }, .. }
+            => {
+                decrease_len(emitter, &mut remaining_len, 1)?;
+                ScalarValue::Int(args_blob.read_i8().expect("already checked len") as i32)
+            },
+
+            | ArgEncoding::Integer { arg0: false, size: 4, format: ast::IntFormat { unsigned: true, radix: _ }, .. }
+            => {
+                decrease_len(emitter, &mut remaining_len, 4)?;
+                ScalarValue::Int(args_blob.read_u32().expect("already checked len") as i32)
+            },
+
+            | ArgEncoding::Integer { arg0: false, size: 2, format: ast::IntFormat { unsigned: true, radix: _ }, .. }
+            => {
+                decrease_len(emitter, &mut remaining_len, 2)?;
+                ScalarValue::Int(args_blob.read_u16().expect("already checked len") as i32)
+            },
+
+            | ArgEncoding::Padding
```

Comment:
I was surprised to see that this loop produces a value for padding.  I'd add a comment saying that the padding values will be filtered out at the end.

Reply:
Good idea. Currently the only reason the values are produced is so that they can be verified to be 0 later on, though I don't think that's happening yet. If there were a way to force a full @blob fallback to happen from inside this function similar to the @mask fallback it's already going to need, then producing the values could be omitted entirely.